### ISSUE TITLE
[Feature request] Include Serious Sam & Half-Life RTX mods

### ISSUE THREAD

**Issue by LTGM** [beeradmoore/dlss-swapper#249]
**Is your feature request related to a problem? Please describe.**

Title. Please include Serious Sam & Half-Life RTX mods

**Comment by jacks-sam1010** [comment]
Hello, we tried to solve the issue.

This is what we did:

Modified GameGridPage.xaml.cs to include Serious Sam & Half-Life RTX mods in the game list. Added a new method to detect and load these mods, and integrated it into the existing game loading process.

You can review changes in this commit: https://github.com/jacks-sam1010/beeradmoore-dlss-swapper-249/commit/ccaaef68da9b6ee7de094e61b74ba1082ef1b6e2.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by beeradmoore** [comment]
Hey @LTGM ,
Thanks for submitting an issue. DLSS Swapper doesn't include individual games, it includes libraries from launchers. 

What launcher did you use to install these, and how is it the mods get installed? I can check why they are not showing up.

**Comment by beeradmoore** [comment]
Thanks for that additional info.

Provided the games are installed by Steam the should show up as their original games (half life or serious sam). Do you see those games as being installed after you restart DLSS Swapper?

Do you see any of your other Steam games showing up?

**Comment by beeradmoore** [comment]
There is a button in the top right that will let you hide games without DLSS.

> I did not replace the original game folder or files.

TL;DR; Next update will support manually adding games.

Yeah, our system looks to see if game library is installed (in this case Steam) and then goes into specific things per platform. In this case it reads what drives you have Steam Libraries on, and then for each library it reads the manifest file for each game on that drive. In that manifest it reads the install directory for that game. It then searches that install directory for a file named `nvngx_dlss.dll`. If it finds it it has it, if it doesnt find it it doesn't. Technically if you copied the dll into a game folder that didn't have it then DLSS Swapper would say it does have it. It does not know anything more than the file is found or not found.

> BTW, , it looks like it will be possible to replace DLSS via NVIDIA's new control panel in the future.

About time :P. They have said they swap out DLSS versions at the driver level for the longs time, but very few if any games actually do it. Next update to DLSS Swapper also supports FSR 3.1, DLSS frame generation, DLSS ray reconstruction, and XeSS. All in the same manner as described above. So as much as I'd like to go work on other things it doesn't look like DLSS Swapper is obsolete yet.
